gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/language/grammar/GenericLiteralVisitor.java

```diff
@@ -506,6 +514,16 @@ public Object visitDateLiteral(final GremlinParser.DateLiteralContext ctx) {
         return DatetimeHelper.parse((String) antlr.argumentVisitor.visitStringArgument(ctx.stringArgument()));
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Object visitUuidLiteral(final GremlinParser.UuidLiteralContext ctx) {
+        if (ctx.stringArgument() == null)
+            return UUID.randomUUID();
+        return UUID.fromString((String) antlr.argumentVisitor.visitStringArgument(ctx.stringArgument()));
```

Comment:
This unchecked cast to a String might not always be safe as we are reworking how variables are getting parsed in the grammar. My suggestion would be to switch stringArgument to stringLiteral (would also need to change this to `genericLiteralVisitor.visitStringLiteral()`) and update the grammar as well (see matching comment in Gremlin.g4).